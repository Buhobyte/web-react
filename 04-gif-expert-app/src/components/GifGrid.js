import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

const GifGrid = ({ category }) => {


    const [count, setCount] = useState(0);

    useEffect(() => {
        dataGifAPI();
    }, [])

    const dataGifAPI = async () => {
        const api_key = 'IeSlcE9JlBLhzpNlK7PPFQHfkvcg0mjY';
        const resp = await fetch('https://api.giphy.com/v1/gifs/search?q=Rick&limit=10&api_key=' + api_key);
        const { data } = await resp.json()

        const dataCatagories = data.map((item) => {
            return {
                id: item.id,
                title: item.title,
                url: item.images?.downsized_medium.url,

            }
        })
        console.log(dataCatagories);
    }

    return (
        <div>
            <h2>{category}</h2>
            <h3>{count}</h3>
            <button onClick={() => { setCount(c => c + 1) }}></button>

        </div>
    )
}

GifGrid.propTypes = {
    categories: PropTypes.array
}

export default GifGrid
