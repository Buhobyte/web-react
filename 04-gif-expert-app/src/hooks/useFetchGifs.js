import { useState } from "react/cjs/react.development"

export const useFectchGifs = () => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    setTimeout(() => {

        setState({
            data: [1, 2, 3, 4, 5, 6],
            loading: false
        })
    },3000)

    return state;
}