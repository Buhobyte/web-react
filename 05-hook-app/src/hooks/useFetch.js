import { useState } from 'react'
import { useEffect } from 'react'

export const useFetch = (url) => {

    const [state, setState] = useState({ error: null, data: '', loading: true, })


    useEffect(() => {

        fetch(url)
            .then(resp => resp.json())
            .then((data) => {

                setState({
                    data,
                    loading: false,
                    error: null
                })
            })
    }, [url])

    return state;

}
