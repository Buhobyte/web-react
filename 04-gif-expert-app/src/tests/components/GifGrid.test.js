import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom'
import GifGrid from '../../components/GifGrid';
import { useFectchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Probar componente add category', () => {

    const category = 'One punch';

    test('deberia mostrar el componente GifGrid', () => {
        useFectchGifs.mockReturnValue({
            data: [], loading: true
        })
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar items cuando se cargan imageenes de usefetchGifs', () => {

        const gifs = [{
            id: 1,
            url: 'http.localhost.com/image1.png',
            title: 'imagen 1'
        }];
        useFectchGifs.mockReturnValue({
            data: gifs, loading: true
        })
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot()

    })


})
