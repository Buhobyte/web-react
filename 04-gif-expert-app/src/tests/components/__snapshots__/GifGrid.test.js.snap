// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar componente add category deberia mostrar el componente GifGrid 1`] = `
<Fragment>
  <h2>
    One punch
  </h2>
  <div
    className="card-grid animate__fadeIn"
  />
</Fragment>
`;
